### PR TITLE
Add swap feature

### PR DESCRIPTION
**Pull request by agrojean-ledger**
Add swap feature to the polkadot app so it can be called from the app exchange with libcalls. 
**Work in progress**

### CODE REVIEW

**Review by lpascal-ledger — COMMENTED** [review]
Sounds like a correct framework

File: app/Makefile [hunk 1]
```diff
@@ -131,7 +131,20 @@ endif
 ############
 # Platform
 
-DEFINES       += PRINTF\(...\)=
+ifndef DEBUG
+	DEBUG = 0
+endif
```
Comment (fbeutin-ledger) [inline app/Makefile:136]: equivalent to 
DEBUG ?= 0

File: app/Makefile [hunk 2]
```diff
@@ -131,7 +131,20 @@ endif
...
+
+ifneq ($(DEBUG),0)
+	DEFINES += HAVE_PRINTF
+	ifeq ($(TARGET_NAME),TARGET_NANOS)
+    	DEFINES += PRINTF=screen_printf
+	else
+    	DEFINES += PRINTF=mcu_usb_printf
+    endif
+else
+    DEFINES += PRINTF\(...\)=
```
Comment (fbeutin-ledger) [inline app/Makefile:146]: Quick note:
Be careful when using the PRINTF macro as its current behavior is a bit dangerous
example:
`PRINTF("%u\n", ++var);`
will change code depending on the DEBUG flag
(PR is open on the SDK to rationalize it)

**Review by fbeutin-ledger — COMMENTED** [review]
Nice start

File: app/Makefile (hunk 2 above)
Comment (agrojean-ledger) [reply app/Makefile:146]: Ok thanks for the tip. I'll be careful.
Do you have a link to the PR ?

File: app/Makefile (hunk 1 above)
Comment (agrojean-ledger) [reply app/Makefile:136]: @fbeutin-ledger I find it more explicit. If you really want I can change it to what you suggest.

File: app/src/swap/handle_check_address.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                               params->address_parameters,
+                               params->address_parameters_length - 1)) // param length - 1B for path length)
+    {
+        PRINTF("Can't parse path !\n");
+        return 0;
+    }
+
+    /* Generate public key from derivation path */
+    uint8_t pub_key[PK_LEN_25519] = {0};
+    crypto_extractPublicKey(kind, path.path, pub_key, PK_LEN_25519);
```
Comment (agrojean-ledger) [inline app/src/swap/handle_check_address.c:40]: This function call seems to cause a segfault in app-exchange... I need to investigate.

File: app/Makefile (hunk 2 above)
Comment (fbeutin-ledger) [reply app/Makefile:146]: No link yet as the SDK will move to another host soon

File: app/Makefile (hunk 1 above)
Comment (fbeutin-ledger) [reply app/Makefile:136]: I prefer `?=` as it's more standard but as you want

File: app/src/swap/handle_check_address.h [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#ifndef _HANDLE_CHECK_ADDRESS_H_
+#define _HANDLE_CHECK_ADDRESS_H_
```
Comment (fbeutin-ledger) [inline app/src/swap/handle_check_address.h:2]: use `#pragma once`, it's less error prone when copying files around
see `app-exchange/src/buffer.h` for example

File: app/Makefile (hunk 1 above)
Comment (agrojean-ledger) [reply app/Makefile:136]: Ok I'll fix it.

File: app/Makefile (hunk 2 above)
Comment (agrojean-ledger) [reply app/Makefile:146]: Ok.

File: app/src/swap/handle_check_address.h (hunk 4 above)
Comment (agrojean-ledger) [reply app/src/swap/handle_check_address.h:2]: Ok I'll fix it.

File: app/Makefile (hunk 1 above)
Comment (agrojean-ledger) [reply app/Makefile]: Done.

File: app/src/swap/handle_check_address.h (hunk 4 above)
Comment (agrojean-ledger) [reply app/src/swap/handle_check_address.h]: Done.

File: app/src/swap/handle_check_address.c (hunk 3 above)
Comment (agrojean-ledger) [reply app/src/swap/handle_check_address.c:40]: Doesn't seem to be the root cause of the bug on the LNS.

**Pull request closed without merging**